### PR TITLE
fix: add helm dependencies with custom CA

### LINKED ISSUE #5596: Custom Certificate Authorities not being respected in Git Applications with Helm dependencies
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

My organization is using a Helm repo (using goharbor) too host custom charts on our network. This Helm repo is using a certificate signed by a non-public CA. When we add the certificate to the ArgoCD certificate store, and create Application specs that are configured to pull directly from Helm, ArgoCD accepts the certificates and is able to retrieve charts without issue.

We're encountering a case where our development teams are creating Git-based Applications that have Helm dependencies on charts in out self-hosted Helm repo. In the case of these apps, when Helm goes to to a dependency update, it rejects the cert from Harbor as being signed by an unknown authority. In this case, whatever certificate injection Argo is doing for a Helm-based app is clearly not being done when resolving Helm dependencies in a Git-based app. Discussed in-depth in [this thread](https://argoproj.slack.com/archives/CASHNF6MS/p1614199573256000)

**To Reproduce**

1. Create a Git repo containing a Helm chart which has a dependency on a chart hosted in a Helm repo with a self-signed certificate
2. Create an Application using that Git repo
3. Watch the errors fly

**Version**

```shell
v1.8.5
```

### PR DESCRIPTION
**Pull request by shubhamagarwal19**
Signed-off-by: Shubham Agarwal <shubhamagarawal19@gmail.com>

Closes: #5596 

Before building the helm dependencies, it fetches the list of repositories specified in Chart.yaml and adds `(helm repo add <repo> --ca-file <ca-file.pem>)` them with the custom CA provided by the user.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: reposerver/repository/repository.go [hunk 1]
```diff
@@ -586,11 +628,23 @@ func helmTemplate(appPath string, repoRoot string, env *v1alpha1.Env, q *apiclie
...
-	h, err := helm.NewHelmApp(appPath, getHelmRepos(q.Repos), isLocal, version)
 
+	repos, err := getHelmDependencyRepos(appPath)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range repos {
+		if !repoExists(r.Repo, q.Repos) {
+			q.Repos = append(q.Repos, repos...)
```
Comment (mayzhang2000) [inline reposerver/repository/repository.go:639]: Should it be append(q.Repos, r) instead of adding all of repos?

**Review by mayzhang2000 — APPROVED** [review]
LGTM

**Comment by shubhamagarwal19** [comment]
> LGTM

@mayzhang2000, thanks for fixing it !!!

**Comment by rbkaspr** [comment]
I saw this merge didn't make the 2.0.1 release window, any chance we'll see this in 2.0.2?

**Pull request merged**